include/profiler.rst

```diff
@@ -0,0 +1,115 @@
+  Set this option to add the Profiler source code to the application.
+  If you use the Event Manager, you can also set this option by selecting :option:`CONFIG_DESKTOP_EVENT_MANAGER_PROFILER_ENABLED`.
+
+Call :cpp:func:`profiler_init` during the application start to initialize the Profiler.
+If you set :option:`CONFIG_DESKTOP_EVENT_MANAGER_PROFILER_ENABLED`, the Profiler is automatically initialized when you initialize the :ref:`event_manager`.
+
+
+Supported backends
+******************
+
+The Profiler supports different backends to visualize the output data.
+Currently, the two supported backends are SEGGER SystemView and a custom backend.
+Both share the same API and communicate with the host using RTT.
+
+
+SEGGER SystemView
+=================
+
+Select this backend to register the Profiler as a middleware module for `SEGGER SystemView`_.
+You can then use a dedicated visualization tool to visualize events.
+
+See the `SEGGER SystemView`_ website for more information.
+
+Set :option:`CONFIG_PROFILER_SYSVIEW` to enable this backend.
+
+Custom backend
+==============
+
+Select the custom backend to use dedicated tools written in Python for event visualization, analysis, and calculating statistics.
+
+To save profiling data, the tools use csv files (for event occurrences) and json files (for event descriptions).
+The scripts can be found under :file:`scripts/profiler/` in the |NCS| folder structure.
+
+Set :option:`CONFIG_PROFILER_NORDIC` to enable this backend.
+
+To use the tools, run the scripts on the command line:
+
+* ``python3 data_collector.py 5 a.csv a.json``
+
+  Connects to the device via RTT, receives profiling data, and saves it to files.
+  As command line arguments, provide the time for collecting data and the files where to save the data.
+
+* ``python3 plot_from_files.py a.csv a.json``
+
+  Plots events from the files that are provided as command line arguments.
+
+* ``python3 real_time_plot.py a.csv a.json``
+
+  Connects to the device via RTT, plots data in real time, and saves the data.
+  As command line arguments, provide the files where to save the data.
+
+
+Visualization
+-------------
+
+When running ``plot_from_files.py`` or ``real_time_plot.py``, the profiled events are visualized in a GUI window.
+
+When displaying Event Manager events, submissions are marked as dots.
+Processing of the events is displayed as rectangles, visualizing the processing time.
+
+Use the **start/stop** button below the plot to pause or resume real time plot translation.
+Scroll to zoom in or out.
+When paused, scrolling zooms to the cursor location.
+When plotting in real time, scrolling zooms to the right edge of the plot.
+Use the middle mouse button to mark an event submission or processing for tracking, and to display the event data.
+
+When plotting is paused, you can click and drag with the left mouse button to pan the plot.
+Click the left and right mouse button simultaneously to place a horizontal line at the cursor location.
```

Comment:
I haven't seen the GUI, but vertical lines would be more common (if time is on the x axis)?

Reply:
Also, is it both buttons simultaneously? The input wasn't clear.

Reply:
Yes, of course. I fixed it.
I was going to take a look at whole documentation after adding profiling custom events section and then ask you for review.